esp-hal/build.rs

```diff
@@ -96,12 +58,6 @@ fn main() -> Result<(), Box<dyn Error>> {
         unreachable!() // We've confirmed exactly one known device was selected
     };
 
-    if detect_atomic_extension("a") || detect_atomic_extension("s32c1i") {
```

Comment:
I just kind of went ahead and removed this check (trying to eliminate calls to `panic!()` here in favour of `error!()`), I think this is probably okay to do? @MabezDev @bjoernQ would appreciate your input though, can revert this change if needed.

Reply:
I guess it's fine since it's already a long time when we removed support. t.b.h. I didn't even know we had that check and I think I haven't seen this trigger for users who had that and asked on Matrix ... but maybe I'm wrong about that

Reply:
Sounds good, its been 3+months, we can just point them to the changelog if they run into the issue - I think it will be pretty obvious from the error message.